gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -199,6 +203,63 @@ def _shard_variants(known_args, pipeline_args, pipeline_mode):
           _GCS_RECURSIVE_WILDCARD]
 
 
+def _get_input_dimensions(known_args, pipeline_args, pipeline_mode):
+  beam_pipeline_options = pipeline_options.PipelineOptions(pipeline_args)
+  google_cloud_options = beam_pipeline_options.view_as(
+      pipeline_options.GoogleCloudOptions)
+
+  estimate_sizes_job_name = pipeline_common.generate_unique_name(
+      _ESTIMATE_SIZES_JOB_NAME)
+  current_job_name = google_cloud_options.job_name
+  if google_cloud_options.job_name:
+    google_cloud_options.job_name += '-' + estimate_sizes_job_name
+  else:
+    google_cloud_options.job_name = estimate_sizes_job_name
+  temp_directory = google_cloud_options.temp_location or tempfile.mkdtemp()
+  temp_estimated_line_count_file_name = '-'.join(
+      [google_cloud_options.job_name,
+       _ESTIMATE_SIZES_FILE_NAME])
+  temp_estimated_line_count_file_path = filesystems.FileSystems.join(
+      temp_directory, temp_estimated_line_count_file_name)
+  with beam.Pipeline(options=beam_pipeline_options) as p:
+    estimates = pipeline_common.get_estimates(
+        p, pipeline_mode, known_args.all_patterns)
+
+    files_size = (estimates
+                  | 'GetFilesSize' >> extract_input_size.GetFilesSize())
+    file_count = (estimates
+                  | 'CountAllFiles' >> beam.combiners.Count.Globally())
+    sample_map = (estimates
+                  | 'ExtractSampleMap' >> extract_input_size.GetSampleMap())
+    estimated_record_count = (sample_map
+                              | extract_input_size.GetEstimatedRecordCount())
+    estimated_sample_count = (sample_map
+                              | extract_input_size.GetEstimatedSampleCount())
+    estimated_line_count = (estimates
+                            | 'GetEstimatedLineCount'
+                            >> extract_input_size.GetEstimatedLineCount())
+    _ = (estimated_line_count
+         | beam.ParDo(extract_input_size.print_estimates_to_file,
+                      beam.pvalue.AsSingleton(estimated_sample_count),
+                      beam.pvalue.AsSingleton(estimated_record_count),
+                      beam.pvalue.AsSingleton(files_size),
+                      beam.pvalue.AsSingleton(file_count),
+                      temp_estimated_line_count_file_path))
+
+  google_cloud_options.job_name = current_job_name
```

Comment:
I don't think this job will change the next pipeline's job name since it will be a new instance from pipeline_options.PipelineOptions(pipeline_args)

Reply:
Hmm, good point. Removed this line and generation of current_job_name one.